File: BubbleSort/BubbleSort.java
public class BubbleSort {
	public static void printData(int[] data) {
		for (int i = 0; i < data.length; i++) {
			System.out.print(data[i] + "  ");
		}
		System.out.println();
	}
	public static void bubbleSort(int[] data) {
		int len = data.length;
		for (int i = 0; i < len; i++) {
			for (int j = 0; j < len - i - 1; j++) {
				if (data[j] > data[j+1]) {
					int temp = data[j];
					data[j] = data[j+1];
					data[j+1] = temp;
				}
				printData(data);
			}
			System.out.println();
		}
	}
	
	public static void main(String[] args) {
		int[] a = {4, 5, 3, 1, 2};
		bubbleSort(a);
		
		printData(a);
	}
}

File: BubbleSort/BubbleSort.py
def bubbleSort(data):
    length = len(data)
    for i in range(length):
        for j in range(length - i - 1):
            if data[j] > data[j + 1]:
                data[j], data[j+1] = data[j+1], data[j]
            print(data)
        print('\n')


data = [4, 5, 3, 2, 1]
bubbleSort(data)

print(data)